src/screens/Node/components/NodeItemDetail/NodeItemDetail.enhanceData.js

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from 'react';
+import ErrorBoundary from '@src/components/ErrorBoundary';
+import {useNavigationParam} from 'react-navigation-hooks';
+import { useSelector } from 'react-redux';
+import {checkLoadingNodeByProductId, getDeviceByProductId} from '@screens/Node/Node.selector';
+import { get } from 'lodash';
+
+const nodeItemDetailEnhanceData = WrappedComp => props => {
+  const onUnstake   = useNavigationParam('onUnstake');
+  const onStake     = useNavigationParam('onStake');
+  const productId   = useNavigationParam('productId');
+  const onImport    = useNavigationParam('onImport');
+  const onWithdraw  = useNavigationParam('onWithdraw');
+  const withdrawTxs = useNavigationParam('withdrawTxs');
+
+  const [processing, setProcessing] = useState(false);
+
+  const device      = useSelector(getDeviceByProductId)(productId);
+  const isLoading   = useSelector(checkLoadingNodeByProductId)(productId);
+  const ip          = device?.Host;
+  const name        = device?.Name;
+  const hasAccount  = !!device?.AccountName;
+  const rewardsList = device?.AllRewards;
+
+  let shouldShowWithdraw = false;
+
+  if (rewardsList && rewardsList.length > 0) {
+    rewardsList.forEach(element => {
+      if (element?.balance > 0) {
+        shouldShowWithdraw = true;
+      }
+    });
```

Comment:
Optimize Performance
```suggestion
    shouldShowWithdraw = rewardsList.some(element => element?.balance > 0);
```

Reply:
thanks for this suggestion